### PR TITLE
Minor cleanup and refactoring

### PR DESCRIPTION
**Pull request by onobc**
@sobychacko this is the lightweight cleanup/refactorings I did. I wanted to get this smaller "noise" out of the way before the subsequent PR which will contain the heavier changes to the template etc..

### CODE REVIEW

File: spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java [hunk 1]
```diff
@@ -188,14 +190,15 @@ private void buildEnhancer() {
 			Map<String, AnnotationEnhancer> enhancersMap =
 					this.applicationContext.getBeansOfType(AnnotationEnhancer.class, false, false);
 			if (enhancersMap.size() > 0) {
+				// BONO: We can simplify this w/ ObjectProvider API (which handles null and also ordering)
```
Comment (onobc) [inline spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java:193]: I will revisit this shortly

File: spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java [hunk 2]
```diff
@@ -227,6 +232,7 @@ public void afterSingletonsInstantiated() {
 		}
 
 		if (this.defaultContainerFactoryBeanName != null) {
+			// BONO: Should we check that its not already set on the registrar (like we do above for the endpointRegistry?)
```
Comment (onobc) [inline spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java:235]: @sobychacko I just saw these comments I had added in the code that would better be served as PR comments. For now, lets just discuss/answer them here and I will remove the comment from the actual code w/ a commit to this PR once we discuss each (several below as well that fit this same bill).

File: spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java [hunk 3]
```diff
@@ -360,9 +354,7 @@ private void processPulsarListenerAnnotation(MethodPulsarListenerEndpoint<?> end
 			endpoint.setAutoStartup(resolveExpressionAsBoolean(autoStartup, "autoStartup"));
```
Comment (onobc) [inline spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java:354]: (had to put this comment here, but the line of interest is https://github.com/spring-projects-experimental/spring-pulsar/pull/24/files#diff-babecbd1e9a205d2937e46186f42982af55c2a50a928add1b6030f4bf06f016cR233)....

```java

		// Set the custom handler method factory once resolved by the configurer
		MessageHandlerMethodFactory handlerMethodFactory = this.registrar.getMessageHandlerMethodFactory();
		if (handlerMethodFactory != null) {
			this.messageHandlerMethodFactory.setHandlerMethodFactory(handlerMethodFactory);
		}
		else {
			addFormatters(this.messageHandlerMethodFactory.defaultFormattingConversionService);
		}
```
It is unclear to me why this is on the ELSE branch (maybe enhance comment above?)

File: spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java [hunk 4]
```diff
@@ -360,9 +354,7 @@ private void processPulsarListenerAnnotation(MethodPulsarListenerEndpoint<?> end
...
 		}
```
Comment (onobc) [inline spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java:355]: (had to put this comment here, but the line of interest is https://github.com/spring-projects-experimental/spring-pulsar/pull/24/files#diff-babecbd1e9a205d2937e46186f42982af55c2a50a928add1b6030f4bf06f016cR249)....
```java
Collection<PulsarListener> classLevelListeners = findListenerAnnotations(targetClass);
```
Looks like we do nothing w/ the class level listeners?

Comment (onobc) [reply spring-pulsar/src/main/java/org/springframework/pulsar/annotation/PulsarListenerAnnotationBeanPostProcessor.java:355]: Soby: We are not using class level listener now - we can remove for now.

**Review by sobychacko — APPROVED** [review]

**Pull request merged**